### PR TITLE
React Play Password Generator

### PR DESCRIPTION
**Pull request by Angryman18**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@Angryman18 Great, could you please open an issue for this one?

**Comment by Angryman18** [comment]
> @Angryman18 Great, could you please open an issue for this one?

how do i do that? im not so good with github yet.

**Comment by atapas** [comment]
> > @Angryman18 Great, could you please open an issue for this one?
> 
> how do i do that? im not so good with github yet.

Added https://github.com/atapas/react-play/issues/144

please add the relevant details.

**Comment by Angryman18** [comment]
> > > @Angryman18 Great, could you please open an issue for this one?
> > 
> > 
> > how do i do that? im not so good with github yet.
> 
> Added #144
> 
> please add the relevant details.

added

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  // generate password button click handler
+  const generateHander = () => {
+    const finalPassword = generatePassword();
+    setPassword({ status: false, password: finalPassword });
+  };
+
+  // handling checkbox and updating the config
+  const handleCheckedItem = (name, checked) => {
+    const modifiedConfig = { ...passwordConfig };
+    delete modifiedConfig.length
```
Comment (atapas) [inline src/plays/password-generator/PasswordGenerator.jsx:80]: In some lines ; are missing. It good to be uniform.

File: src/plays/password-generator/Readme.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Password Generator
```
Comment (atapas) [inline src/plays/password-generator/Readme.md:1]: Put some explanation in this. See other play readmes as reference.

File: src/plays/password-generator/password-generator-style.css [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    padding: 0;
+    margin: 0;
+    box-sizing: border-box;
+}
+
+.title {
+    text-align: center;
+}
+
+section {
```
Comment (atapas) [inline src/plays/password-generator/password-generator-style.css:13]: Do not override the selectors directly. Always scope it to your play with a class name. The same is applicable to other selectors.

**Review by atapas — CHANGES_REQUESTED** [review]
@Angryman18 Thanks for adding this play. Very neat. I have a few review comments. Once you take care, we are good to go.

File: src/plays/password-generator/PasswordGenerator.jsx (hunk 1 above)
Comment (Angryman18) [reply src/plays/password-generator/PasswordGenerator.jsx]: yes. semicolons are added

File: src/plays/password-generator/Readme.md (hunk 2 above)
Comment (Angryman18) [reply src/plays/password-generator/Readme.md]: yes the explanations are added.

**Review by atapas — APPROVED** [review]
@Angryman18 Yayy, Looks good.

**Comment by atapas** [comment]
@all-contributors please add @Angryman18  for Code

**Comment by atapas** [comment]
@Angryman18 Thanks for your first contribution to the project. Looking for more to come, yay 😀.

If you are planning to share your contribution experience anywhere publicly, please tag our Twitter handle(https://twitter.com/ReactPlayIO). Also, don't be shy to support the project with a star. Keep going.

**Pull request merged**